va/validation-authority.go

```diff
@@ -398,108 +360,6 @@ func (va *ValidationAuthorityImpl) validateTLSWithZName(identifier core.AcmeIden
 	return challenge, challenge.Error
 }
 
-// TODO(https://github.com/letsencrypt/boulder/issues/894): Delete this method
-func (va *ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentifier, input core.Challenge) (core.Challenge, error) {
-	challenge := input
-
-	if identifier.Type != core.IdentifierDNS {
-		challenge.Status = core.StatusInvalid
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.MalformedProblem,
-			Detail: "Identifier type for SimpleHTTP was not DNS",
-		}
-
-		va.log.Debug(fmt.Sprintf("SimpleHTTP [%s] Identifier failure", identifier))
-		return challenge, challenge.Error
-	}
-
-	// Perform the fetch
-	path := fmt.Sprintf(".well-known/acme-challenge/%s", challenge.Token)
-	useTLS := (challenge.TLS == nil) || *challenge.TLS
-	body, challenge, err := va.fetchHTTP(identifier, path, useTLS, challenge)
-	if err != nil {
-		return challenge, err
-	}
-
-	// Parse and verify JWS
-	parsedJws, err := jose.ParseSigned(string(body))
-	if err != nil {
-		err = fmt.Errorf("Validation response failed to parse as JWS: %s", err.Error())
-		va.log.Debug(err.Error())
-		challenge.Status = core.StatusInvalid
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.UnauthorizedProblem,
-			Detail: err.Error(),
-		}
-		return challenge, err
-	}
-
-	// Check that JWS body is as expected
-	// * "type" == "simpleHttp"
-	// * "token" == challenge.token
-	// * "tls" == challenge.tls || true
-	target := map[string]interface{}{
-		"type":  core.ChallengeTypeSimpleHTTP,
-		"token": challenge.Token,
-		"tls":   (challenge.TLS == nil) || *challenge.TLS,
-	}
-	err = verifyValidationJWS(parsedJws, challenge.AccountKey, target)
-	if err != nil {
-		va.log.Debug(err.Error())
-		challenge.Status = core.StatusInvalid
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.UnauthorizedProblem,
-			Detail: err.Error(),
-		}
-		return challenge, err
-	}
-
-	challenge.Status = core.StatusValid
-	return challenge, nil
-}
-
-// TODO(https://github.com/letsencrypt/boulder/issues/894): Delete this method
-func (va *ValidationAuthorityImpl) validateDvsni(identifier core.AcmeIdentifier, input core.Challenge) (core.Challenge, error) {
```

Comment:
Unfortunately, there's a race here at deploy time. The other systems might be emitting simpleHTTP and DVSNI challenges that the VA is expected to understand. Similarly, clients could have elected to use those challenges, too.

So, first, we need this PR to remove the places where the challenges are generated for the client to select from, then give it a little bit before we disallow them from the VA completely.